go/vt/worker/legacy_row_splitter.go

```diff
@@ -54,17 +58,34 @@ func (rs *RowSplitter) StartSplit() [][][]sqltypes.Value {
 
 // Split will split the rows into subset for each distribution
 func (rs *RowSplitter) Split(result [][][]sqltypes.Value, rows [][]sqltypes.Value) error {
+	droppedRows := 0
+	droppedKeys := []string{}
+
+	defer func() {
```

Comment:
This looks good out, but out of curiosity: what's the rationale of using a defer here?  Seems like it doesn't fit the normal pattern where I would use a defer (e.g the classic lock/unlock of mutexes)

Reply:
we accumulate the number of dropped rows so that we don't lock the mutex a ton but that means we need to report it at the end of the function.

If we hit a bad row we bail immediately though and I didn't want to have a more complicated approach to ensure that we counted dropped rows on both the "function completed" and "function errored bc of a bad row mapping" scenario